tests/metrics/functional/test_image_gradients.py

```diff
@@ -0,0 +1,253 @@
+    HEIGHT = 5
+    WIDTH = 5
+    CHANNELS = 1
+
+    device = torch.device("cpu")
+
+    image = torch.arange(0, BATCH_SIZE * HEIGHT * WIDTH * CHANNELS, dtype=torch.float32, device=device)
+    image = torch.reshape(image, (HEIGHT, WIDTH))
+
+    with pytest.raises(RuntimeError):
+        image_gradients(image, device)
+
+
+def test_multi_batch_image_gradients():
+    """Test whether the module correctly calculates gradients for known input
+    with non-unity batch size.Example input-output pair taken from TF's implementation of i
+    mage-gradients
+    """
+
+    BATCH_SIZE = 5
+    HEIGHT = 5
+    WIDTH = 5
+    CHANNELS = 1
+
+    device = torch.device("cpu")
+
+    single_channel_img = torch.arange(0, 1 * HEIGHT * WIDTH * CHANNELS, dtype=torch.float32, device=device)
+    single_channel_img = torch.reshape(single_channel_img, (CHANNELS, HEIGHT, WIDTH))
+    image = torch.stack([single_channel_img for _ in range(BATCH_SIZE)], dim=0)
+
+    true_dy = [
+        [
+            5.0,
+            5.0,
+            5.0,
+            5.0,
+            5.0,
+        ],
+        [
+            5.0,
+            5.0,
+            5.0,
+            5.0,
+            5.0,
+        ],
+        [
+            5.0,
+            5.0,
+            5.0,
+            5.0,
+            5.0,
+        ],
+        [
+            5.0,
+            5.0,
+            5.0,
+            5.0,
+            5.0,
+        ],
+        [
+            0.0,
+            0.0,
+            0.0,
+            0.0,
+            0.0,
+        ],
+    ]
+
+    true_dx = [
+        [
+            1.0,
+            1.0,
+            1.0,
+            1.0,
+            0.0,
+        ],
+        [
+            1.0,
+            1.0,
+            1.0,
+            1.0,
+            0.0,
+        ],
+        [
+            1.0,
+            1.0,
+            1.0,
+            1.0,
+            0.0,
+        ],
+        [
+            1.0,
+            1.0,
+            1.0,
+            1.0,
+            0.0,
+        ],
+        [
+            1.0,
+            1.0,
+            1.0,
+            1.0,
+            0.0,
+        ],
+    ]
+
+    true_dy = torch.Tensor(true_dy)
+    true_dx = torch.Tensor(true_dx)
+
+    dy, dx = image_gradients(image, device)
+
+    for batch_id in range(BATCH_SIZE):
+        assert torch.allclose(dy[batch_id, 0, :, :], true_dy)
+    assert dy.shape == (BATCH_SIZE, 1, HEIGHT, WIDTH)
+    assert dx.shape == (BATCH_SIZE, 1, HEIGHT, WIDTH)
+
+
+def test_image_gradients():
+    """Test whether the module correctly calculates gradients for known input.
+    Example input-output pair taken from TF's implementation of image-gradients
+    """
+
+    BATCH_SIZE = 1
+    HEIGHT = 5
+    WIDTH = 5
+    CHANNELS = 1
+
+    device = torch.device("cpu")
+
+    image = torch.arange(0, BATCH_SIZE * HEIGHT * WIDTH * CHANNELS, dtype=torch.float32, device=device)
+    image = torch.reshape(image, (BATCH_SIZE, CHANNELS, HEIGHT, WIDTH))
+
+    true_dy = [
```

Comment:
same here

Reply:
Sure.

Reply:
made the recommended changes in the latest commit